homeassistant/components/nest/api.py

```diff
@@ -114,9 +114,14 @@ async def new_subscriber(
         implementation, config_entry_oauth2_flow.LocalOAuth2Implementation
     ):
         raise TypeError(f"Unexpected auth implementation {implementation}")
-    subscription_name = entry.data.get(
-        CONF_SUBSCRIPTION_NAME, entry.data[CONF_SUBSCRIBER_ID]
-    )
+    if not (
+        subscription_name := entry.data.get(
+            CONF_SUBSCRIPTION_NAME, entry.data.get(CONF_SUBSCRIBER_ID)
+        )
+    ):
+        raise ValueError(
```

Comment:
What happens when this is raised? How can the user recover from this?

Reply:
This is a shouldn't happen / bug.  Let me try one more simpler way...